libraries/helpers.rb

```diff
@@ -307,5 +307,25 @@ def package_options
 
       options
     end
+
+    #
+    # Checks the deprecated properties of chef-ingredient and prints warning
+    # messages if any of them are being used.
+    #
+    def check_deprecated_properties
```

Comment:
What would you like to happen if a user uses a `chef-*` or `opscode-*` prefixed product name? Without some logic like I mentioned above their CCR will just fail with an unfriendly message right? At the very least you need to validate what they provide against `PRODUCT_MATRIX.products` to ensure a valid key has been given. A friendlier approach would removing the prefix and printing the warning.